### PR TITLE
WIP[artifactory] Support custom CA and certificates

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
Support custom CA and certificates (work in progress)

### CODE REVIEW

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 1]
```diff
@@ -120,6 +123,22 @@ spec:
           cp -fv /tmp/etc/system.yaml {{ .Values.artifactory.persistence.mountPath }}/etc/system.yaml;
           echo "Remove {{ .Values.artifactory.persistence.mountPath }}/lost+found folder if exists";
           rm -rfv {{ .Values.artifactory.persistence.mountPath }}/lost+found;
+        {{- if .Values.access.tlsEnabled }}
+          echo "Copy access.config.latest.yml to {{ .Values.artifactory.persistence.mountPath }}/etc";
+          mkdir -p {{ .Values.artifactory.persistence.mountPath }}/etc/access;
+          cp -fv /tmp/etc/access.config.import.yml {{ .Values.artifactory.persistence.mountPath }}/etc/access/access.config.import.yml;
+        {{- end }}
+        {{- if .Values.access.resetAccessCAKeys }}
+          echo "ResetAccessCAKeys";
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-statefulset.yaml:132]: `echo "Resetting Access CA Keys";`

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 2]
```diff
@@ -120,6 +123,22 @@ spec:
...
+          mkdir -p {{ .Values.artifactory.persistence.mountPath }}/bootstrap/etc/access/keys;
+          touch {{ .Values.artifactory.persistence.mountPath }}/bootstrap/etc/access/keys/reset_ca_keys;
+        {{- end }}
+        {{- if .Values.access.customCertificatesSecretName }}
+          echo "Copy certs to {{ .Values.artifactory.persistence.mountPath }}/bootstrap/etc/access/keys";
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-statefulset.yaml:137]: `echo "Copying custom certificates to`

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 3]
```diff
@@ -151,6 +170,27 @@ spec:
...
+        {{- end }}
+        {{- if .Values.access.customCertificatesSecretName }}
+        - name: access-certs
+          mountPath: "/tmp/etc/tls.crt"
+          subPath: tls.crt
+        - name: access-certs
+          mountPath: "/tmp/etc/tls.key"
+          subPath: tls.key
+        {{- end }}
+        {{- if .Values.access.customCertificatesSecretName }}
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-statefulset.yaml:186]: Duplicate...

File: stable/artifactory/templates/artifactory-access-config.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  name: {{ template "artifactory.fullname" . }}-access-config
+  labels:
+    app: {{ template "artifactory.name" . }}
+    chart: {{ template "artifactory.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+type: Opaque
+stringData:
+  access.config.import.yml: |
+    security:
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-access-config.yaml:14]: I suggested getting the value from values.yaml, which will support more config in the future.

Comment (chukka) [reply stable/artifactory/templates/artifactory-access-config.yaml]: Done

File: stable/artifactory/templates/artifactory-statefulset.yaml (hunk 1 above)
Comment (chukka) [reply stable/artifactory/templates/artifactory-statefulset.yaml]: Done

File: stable/artifactory/templates/artifactory-statefulset.yaml (hunk 2 above)
Comment (chukka) [reply stable/artifactory/templates/artifactory-statefulset.yaml]: Done

**Pull request closed without merging**